pkg/http-client/httpclient.go

```diff
@@ -13,15 +13,85 @@ limitations under the License.
 package httpclient
 
 import (
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
+	"time"
+)
+
+const (
+	defaultTimeoutSeconds = 10
 )
 
 type Client interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
+//
+// creates a new instance of Client, given a path to addtional certificates
+// certFile may be empty string, which means no additional certs will be used
+//
+func New() *http.Client {
+	// Return Transport for testing purposes
+	return &http.Client{
+		Timeout: time.Second * defaultTimeoutSeconds,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+	}
+}
+
+//
+// creates a new instance of Client, given a path to addtional certificates
+// certFile may be empty string, which means no additional certs will be used
+//
```

Comment:
Just note that generally golang docstrings are not preceded with an empty `//\n` line (nor followed by one), and usually begin with the function/struct/constant name, such as:
```
// Foo does stuff before returning nil
// Extra details about Foo here.
func Foo() nil
```
See https://blog.golang.org/godoc for the background, but it just plays well with the `godoc` binary.

Reply:
mmm, maybe, but it makes code easier to read. Oh well, when in Rome 👍 I will change it